Core/src/org/sleuthkit/autopsy/datamodel/ContentUtils.java

```diff
@@ -404,22 +407,24 @@ public Void visitDir(AbstractFile dir) {
                 int numProcessed = 0;
                 // recurse on children
                 for (Content child : dir.getChildren()) {
-                    java.io.File childFile = getFsContentDest(child);
-                    ExtractFscContentVisitor<T, V> childVisitor
-                            = new ExtractFscContentVisitor<>(childFile, progress, worker, false);
-                    // If this is the source directory of an extract it
-                    // will have a progress and worker, and will keep track
-                    // of the progress bar's progress
-                    if (worker != null && worker.isCancelled()) {
-                        break;
+                    if (child instanceof FsContent){
```

Comment:
You were right the first time, and I misled you. This instanceof check is not quite correct. The visitor is supposed to be able to "extract"  LayoutFile,  DerrivedFile, LocalFile, LocalFile and recurse through VirtualDirectory and LocalDirectory. All of these classes extend AbstractFile, not FsContent.